pkg/policy/types/policyentry.go

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package types
+
+import (
+	"github.com/cilium/cilium/pkg/labels"
+	"github.com/cilium/cilium/pkg/policy/api"
+)
+
+// PolicyEntry specifies the L3/L4 details of a single policy rule
+//
+// +deepequal-gen=true
+// +deepequal-gen:private-method=true
+type PolicyEntry struct {
+	// Authentication specifies the cryptographic authentication required for the traffic to be allowed
+	Authentication *api.Authentication
+
+	// Log specifies custom policy-specific Hubble logging configuration.
+	Log api.LogConfig
+
+	// EndpointSelector specifies the endpoint that this rule applies to
+	EndpointSelector api.EndpointSelector
+
+	// L3 specifies the source/destination endpoints or all endpoints if empty
+	//
+	// +deepequal-gen=false
+	L3 EndpointSelectorInterfaceSlice
+
+	// L4 specifies the source/destination port rules or none if empty
+	L4 api.PortRules
+
+	// Labels stores optional metadata.
+	Labels labels.LabelArray
+
+	// DefaultDeny is true if affected subjects should have non-selected traffic denied
+	DefaultDeny bool
+
+	// Deny is true if this rule should deny traffic
+	Deny bool
+
+	// Ingress is true if rule should affect ingress traffic, false otherwise
+	Ingress bool
+
+	// Node is true if EndpointSelector refers to a node
+	Node bool
+}
+
+func (in *PolicyEntry) DeepEqual(other *PolicyEntry) bool {
+	if other == nil {
+		return false
+	}
+	if len(in.L3) != len(other.L3) {
+		return false
+	}
+	for i := range in.L3 {
+		switch v1 := in.L3[i].(type) {
+		case api.EndpointSelector:
+			if v2, ok := other.L3[i].(api.EndpointSelector); !ok || !v1.DeepEqual(&v2) {
+				return false
+			}
+		case api.CIDR:
+			if v2, ok := other.L3[i].(api.CIDR); !ok || v1 != v2 {
+				return false
+			}
+		case api.CIDRRule:
+			if v2, ok := other.L3[i].(api.CIDRRule); !ok || !v1.DeepEqual(&v2) {
+				return false
+			}
+		default:
+			return false
+		}
+	}
+	return in.deepEqual(other)
+}
+
+// PolicyEntries is a slice of pointers to PolicyEntry
+type PolicyEntries []*PolicyEntry
+
+// EndpointSelectorInterface is a generic representation of an endpoint selector.
+// It can be converted to an EndpointSelector or FQDNSelector.
+type EndpointSelectorInterface interface {
```

Comment:
It seems like this is always used to select peers, maybe it would be simpler to refer to this as:
```suggestion
// PeerSelector is a generic representation of a selector for network peers in a policy rule.
type PeerSelector interface {
```

Reply:
(I also dropped the `It can be converted ...` since that seems like it's already an out-of-date or not comprehensive comment)

Reply:
Thank you for this suggestion. As you can see, I was struggling with naming. I like PeerSelector, it is short and concise. I have added a commit that changes the name.